test/builder/dummies.go

```diff
@@ -549,23 +550,21 @@ func DummyVirtualMachineWebConsoleRequest(namespace, wcrName, vmName, pubKey str
 
 func DummyVirtualMachineSnapshot(namespace, name, vmName string) *vmopv1.VirtualMachineSnapshot {
 	return &vmopv1.VirtualMachineSnapshot{
-		TypeMeta: metav1.TypeMeta{
```

Comment:
I believe these are also needed for other tests

Reply:
I see DummyBasicVirtualMachine also defines this https://github.com/vmware-tanzu/vm-operator/blob/a173780248ce5ffd1a41b5e2979fa703c8c9e56e/test/builder/dummies.go#L284-L372.
Defining this makes it convenient when we try to create LocalObjectRef directly from the object using function like `vmSnapshotCRToLocalObjectRefWithDefaultVersion` 
https://github.com/vmware-tanzu/vm-operator/blob/main/controllers/virtualmachinesnapshot/virtualmachinesnapshot_controller_unit_test.go#L436
Since I'm comparing the whole LocalObjectRef instead of just comparing the object name